tuning/data/data_handlers.py

```diff
@@ -257,6 +257,37 @@ def apply_tokenizer_chat_template(
     }
 
 
+def tokenize(
+    element: Union[Dict[str, str], Dict[str, List]],
+    tokenizer: AutoTokenizer,
+    dataset_text_field: str,
+    truncation: Union[bool, str] = None,
+    max_length: int = None,
+    **kwargs,
+):
+    """Function (data handler) to tokenize dataset columns.
+       Expects to be run as a HF Map API function.
+    Args:
+        element: the HF Dataset element.
+        tokenizer: Tokenizer to be used.
+        dataset_text_field: the dataset field to tokenize
+        truncation: Truncation strategy to use, refer the link
+                    (https://huggingface.co/docs/transformers/en/pad_truncation)
+        max_length: Max length to truncate the samples to.
+        kwargs: Any additional kwargs that need to be passed to the tokenizer can be passed as
+                kwargs['tokenizer_kwargs']
+    Returns:
+        tokenized dataset elemenent field "dataset_text_field"
+    """
+    tokenizer_kwargs = kwargs.get("tokenizer_kwargs", {})
+    return tokenizer(
```

Comment:
Will the tokenizer be initialized for every invocation? is there no way to create a tokenizer object before hand, and reuse it?

Reply:
@ashokponkumar  this is the tokenizer call...we initialize it in the data preprocessor backend its an argument to this function...did you mean tokenizer_kwargs? if so I can make them as part of function args